src/optionals/system.c

```diff
@@ -281,6 +281,77 @@ static Value exitNative(DictuVM *vm, int argCount, Value *args) {
     return EMPTY_VAL; /* satisfy the tcc compiler */
 }
 
+static Value chdirNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 1) {
+        runtimeError(vm, "chdir() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0])) {
+        runtimeError(vm, "chdir() argument must be a string");
+        return EMPTY_VAL;
+    }
+
+    char *dir = AS_CSTRING(args[0]);
+
+    int retval = chdir(dir);
+
+    if (retval < 0) {
+        ERROR_RESULT;
+    }
+
+    return newResultSuccess(vm, NIL_VAL);
+}
+
+static Value chmodNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 2) {
+        runtimeError(vm, "chmod() takes 2 arguments (%d given).", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0]) || (!IS_STRING(args[1]) && !IS_NIL(args[1]))) {
+        runtimeError(vm, "first chmod() argument must be a string.");
+        return EMPTY_VAL;
+    }
+
+    ObjString *file = AS_STRING(args[0]);
+    ObjString *mode = AS_STRING(args[1]);
+
+    int i = strtol(mode->chars, 0, 8);
+
+    if (chmod(file->chars, i) == -1) {
+        char errMsg[1024];
```

Comment:
For these parts, can the error length be greater than 256? Do we need to increase this? Just curious why we can't use the `ERROR_RESULT` macro here!